cmd/build.go

```diff
@@ -99,7 +101,74 @@ func doBuild(ctx *cli.Context) error {
 	if err != nil {
 		return err
 	}
+	if !args.Config.Userns {
+		binary, err := os.Readlink("/proc/self/exe")
+		if err != nil {
+			return err
+		}
 
+		cmd := []string{
+			binary,
+			"--oci-dir", args.Config.OCIDir,
+			"--roots-dir", args.Config.RootFSDir,
+			"--stacker-dir", args.Config.StackerDir,
+			"--storage-type", args.Config.StorageType,
+		}
+
+		if args.Config.Debug {
+			cmd = append(cmd, "--debug")
+		}
+
+		if ctx.GlobalBool("no-progress") {
+			cmd = append(cmd, "--no-progress")
+		}
+
+		if ctx.GlobalBool("progress") {
+			cmd = append(cmd, "--progress")
+		}
+
+		cmd = append(cmd, "--internal-userns")
+		cmd = append(cmd, "build")
+
+		if args.LeaveUnladen {
+			cmd = append(cmd, "--leave-unladen")
+		}
+
+		if args.NoCache {
+			cmd = append(cmd, "no-cache")
+		}
+
+		if len(args.OnRunFailure) > 0 {
+			cmd = append(cmd, "--on-run-failure")
+			cmd = append(cmd, args.OnRunFailure)
+		}
+
+		if args.OrderOnly {
+			cmd = append(cmd, "--order-only")
+		}
+
+		if args.OrderOnly {
+			cmd = append(cmd, "--order-only")
+		}
+
+		for _, arg := range args.Substitute {
+			cmd = append(cmd, "--substitute")
+			cmd = append(cmd, arg)
+		}
+
+		for _, arg := range args.LayerTypes {
+			cmd = append(cmd, "--layer-type")
+			cmd = append(cmd, string(arg))
+		}
+
+		if len(ctx.String("stacker-file")) > 0 {
+			cmd = append(cmd, "-f")
+			cmd = append(cmd, ctx.String("stacker-file"))
+		}
+
+		err = container.MaybeRunInUserns(cmd, "Build in container")
+		return err
+	}
 	builder := stacker.NewBuilder(&args)
 	return builder.BuildMultiple([]string{ctx.String("stacker-file")})
```

Comment:
How does this work if someone invokes recursive-build?

Reply:
Yes, I forgot about recursive builds. Thank you!